Collections

spficklin opened the pull request:
This pull request is to merge in code to support multi-site data collections, including creation of downloadable files (CSV, TAB, FASTA) that are mashups of data from multiple sites.

The file creations are really slow still and that needs looking into.  It would take days to build a data collection with 100,000 remote genes.  That's not good enough, but it is functional as far as I have tested.

shawnawsu committed:
start of data collections storing remotely so I do not overwrite it by accident for a third time

shawnawsu committed:
admin interface updates for the collections functionality

shawnawsu committed:
working adding process but still working on the file download button

shawnawsu committed:
changes for the collections before adding the web services api hooks

shawnawsu committed:
tripal web services api calls for adding and removing sites

spficklin committed:
Merged in 7.x-3.x and adjusted the update functions

shawnawsu committed:
updated the tripal collection bundle schema to remove the unique constraint on the collection id

shawnawsu committed:
updated the name of the add function updated comments and added delete for bundle collection table

shawnawsu committed:
file construciton when 2 or more bundles is specified is now working as is the links on the dashboard

shawnawsu committed:
adding headers to single file download csv files

shawnawsu committed:
it would help if I had the correct table name to delete the collection bundle

spficklin committed:
Altered bundle class to have accession available

spficklin committed:
Merge branch 'collections' of github.com:tripal/tripal into collections

spficklin committed:
Fixes to bundle class

spficklin committed:
Merge branch '7.x-3.x' into collections

spficklin committed:
Updated web services to indicate the Tripal formatters that a field can support

shawnawsu committed:
at the final hurdle for remote fields in data collections just need to get the field data to write to the file

shawnawsu committed:
still working on the field generation for the data collection but web services integration is mostly complete

shawnawsu committed:
getting the fields to write to the file but I have introduced an error I have spent most of the day trying to run down fun times.

shawnawsu committed:
missed this file in the most recent commit

shawnawsu committed:
mergeing in the most recent changes to the master branch and fixing a merge conflict

shawnawsu committed:
breaking down the redunant bits into other functions cleaning up comments and getting the getHeader to work

spficklin committed:
Fixed merge conflict

spficklin committed:
Fixed merge conflict

spficklin committed:
Fixed issue #285

spficklin committed:
Added some improved failure messages for terms

shawnawsu committed:
refactored all the downloader files to clean them up and remove the stupid number of if else statements I had through moving to setting the collection bundle as an accessible variable

shawnawsu committed:
Merge branch 'collections' of github.com:tripal/tripal into collections

spficklin committed:
Changed the way TripalFields specify their data formatters

shawnawsu committed:
adding the fix so that content ends up in the correct column adn the header has single values

shawnawsu committed:
Merge branch 'collections' of github.com:tripal/tripal into collections

shawnawsu committed:
Merge branch '7.x-3.x' into collections

shawnawsu committed:
update to change the header functionality and the processing of the tabs within the format entity

spficklin committed:
Fixed issues with web services that affected collections

spficklin committed:
Fixed leftover bugs from yesterdays changes. Fixed FASTA downloaders

spficklin committed:
Added progress indicator to the TripalJob class:

spficklin committed:
Adjusted how files are constructed to improve performance.

spficklin committed:
Moved WS related functions from collections over to WS API. Working on performance improvements

spficklin committed:
Testing schema:description for summary of structured value in web services

spficklin committed:
Fixed minor typo in formatters

spficklin committed:
Added the ability with webservices to specify exactly the entity IDs to return in bulk, as well as the fields that should be present

spficklin committed:
Added cache clear function for web service caches

spficklin committed:
Merge branch '7.x-3.x' into collections

spficklin committed:
Whoops fixed some typos

spficklin committed:
Working on speeding remote queries

spficklin committed:
Fixed bugs in _tripal_update_remote_entity_field

spficklin commented:
Some further info... it took approximately 3 minutes to create a data collection of about 5000 "cached" genes local to my machine.  If the records are not cached it takes longer....

bradfordcondon commented:
* The "contains" parameter is case sensitive by default.  Not a big fan.
* I currently end up with empty collections with no supported file types creating collections. Let's hold off on merging until we can atl east determine if this is an error on my end.

bradfordcondon committed:
minor changes and a base test for collections

spficklin committed:
Fixed bug with creating collection in views

spficklin commented:
What 'contains' parameter do you mean @bradfordcondon

spficklin committed:
Fixed classes functions that were improperly defined after notice via PHP 7.2 upgrade.

spficklin commented:
Okay, I fixed the issue with collections not being created with the Data Search Views.  Also, I added some code to the unit test case to get it to at least run, but we really should plan out how to organize these tests.  I was going to write a few tests for data collections but I have to have data in the site and so we'd need to pre-populate a dataset.  So, I think we need to write unit tests for the GFF3 and FASTA loaders first.  Once that's done then we can use that data loaded to do the unit tests on data collections.

spficklin committed:
Trying to revert test so Travis CI completes

bradfordcondon commented:
>contains parameter

shoot im sorry i realize tahts for the site search, not related to collections.


>testing organization and datasets

First of all I totally agree we need to discuss our plan for organizing tests.

Regarding data for tests.  I had originally envisioned using dockers pre-populated with data as an "official test docker".  However, I've come to believe that best practice is for each test suite is to insert and remove test data in the buildUp/tearDown (which gets run at the start/end of each test in the class).  Because you dont know what order tests will run in, relying on data inserted by a different class isn't great.  Additionally, if a FASTA loader didnt remove features the first run through, then the test would probably fail the second run through because the features were already in the database.  This wouldnt happen on Travis, but it would happen for local testing.

Anyway obviously we need to discuss more and we'll do so elsewhere.

Thanks for pushing changes I'll test again.

bradfordcondon commented on tripal/includes/TripalFieldDownloaders/TripalFieldDownloader.inc:
```diff
@@ -4,67 +4,159 @@
+    // Set the collection ID of the collection that this downloader will use.
+    $this->collection_id = $collection_id;
+    $this->outfile = $user_dir . '/' . $outfile_name;
+    $this->selected_fields = $selected_fields;
```
`$selected_fields` not defined

bradfordcondon commented on tripal/includes/TripalFieldDownloaders/TripalFieldDownloader.inc:
```diff
@@ -76,7 +168,36 @@ abstract class TripalFieldDownloader {
+    // If a field is a TripalField then check its supported downloaders.
+    if (tripal_load_include_field_class($field_type)) {
+      $formatters = $field_type::$download_formatters;
+      if (in_array($formatter, $settings['download_formatters'])) {
```
`$formatter`, `$settings` not defined?

bradfordcondon commented on tripal/includes/TripalFieldDownloaders/TripalFieldDownloader.inc:
```diff
@@ -91,58 +212,156 @@ abstract class TripalFieldDownloader {
+    }
+
+    if ($job) {
+      $job->setItemsHandled($num_handled);
```
`$num_handled` not defined?

bradfordcondon commented on tripal/includes/TripalFieldDownloaders/TripalFieldDownloader.inc:
```diff
@@ -152,6 +371,113 @@ abstract class TripalFieldDownloader {
+          if (!array_key_exists($accession, $this->printable_fields)) {
+            // Only include fields that support this downloader type in
+            // or list of printable fields.
+            if ($this->isFieldSupported($field, $instance)) {
```
$field not defined.  `$site` maybe?

bradfordcondon committed:
fix variables presumably not ported from refactoring

spficklin committed:
Fixed bugs deleting collections and warnings from PHP 7.2 upgrade

spficklin committed:
Oops mised this item pointed out by bradford

spficklin commented:
Okay, 3rd time's the charm.... Fixed issues with messages on deleting a collection, PHP 7.2 warnings and the items pointed out by @bradfordcondon

spficklin committed:
Commented out test until we get something more established (i.e. preloaded data)

spficklin committed:
removed the TripalEntityCollection test until we get testing issues resolved and so the pull request will pass TravisCI

bradfordcondon commented on tripal/includes/TripalEntityCollection.inc:
```diff
@@ -400,73 +455,124 @@ class TripalEntityCollection {
           }
         }
+        if ($job) {
+          $job->setItemsHandled($num_handled + count($slice));
```
I don't think this $num_handled is defined

bradfordcondon reviewed (approved):
tested local collections a bunch and it all looks good!

spficklin committed:
Merge branch '7.x-3.x' into collections
